docs/docs/programmatically-create-pages-from-data.md

```diff
@@ -53,7 +53,9 @@ For each page we want to create we must specify the `path` for visiting that
 page, the `component` template used to render that page, and any `context`
```

Comment:
What do you think of the following sentence?

For each page we want to create, we must specify the path for visiting that page, the component template used to render that page, and any context we need to render the component. The context parameter is optional though often times it will include a unique identifier that can be used to query for associated data that will be used inside your page. All context values are made available in the page query as arguments prefaced with $ (ex. slug -> $slug).
```js
export const query = graphql`
  query($slug: String!) {	  query($slug: String!) {
  }
`
```

I'm not sure if we need to show an example as it's a few lines below anyway.

cc @gillkyle

Reply:
See my comments [here](https://github.com/gatsbyjs/gatsby/pull/15984#pullrequestreview-265458577), I think this would be great 🙂